snappy/overlord.go

```diff
@@ -218,6 +191,30 @@ func ActivateSnap(s *Snap, inter interacter) error {
 		return err
 	}
 
+	return nil
+}
+
+// RemoveGeneratedWrappers removes the generated services, binaries, desktop
+// wrappers
+func RemoveGeneratedWrappers(s *Snap, inter interacter) error {
+	if err := removePackageBinaries(s.m, s.basedir); err != nil {
+		logger.Noticef("Failed to remove binaries for %q: %v", s.Name(), err)
+	}
+
+	if err := removePackageServices(s.m, s.basedir, inter); err != nil {
+		logger.Noticef("Failed to remove services for %q: %v", s.Name(), err)
+	}
+
+	if err := removePackageDesktopFiles(s.m); err != nil {
+		logger.Noticef("Failed to remove desktop files for %q: %v", s.Name(), err)
+	}
+
+	return nil
```

Comment:
It's strange that this returns an error that can never be set, even when it fails altogether.